asr-python-advanced-tao-ngram-deployment.ipynb

```diff
@@ -0,0 +1,497 @@
+   "source": [
+    "Run the following command to build an `.rmir` file from an `.arpa`-formatted n-gram language model file:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Syntax: \n",
+    "# riva-build <task-name> \\\n",
+    "#     output-dir-for-rmir/model.rmir:key \\\n",
+    "#     dir-for-riva/acoustic_model.riva:key \\\n",
+    "#     --decoding_language_model_arpa=lm_model.arpa\n",
+    "! docker run --rm --gpus 0 -v $MODEL_LOC:/servicemaker-dev $RIVA_SM_CONTAINER -- \\\n",
+    "    riva-build speech_recognition \\\n",
+    "        /servicemaker-dev/asr_arpa_ngram_lm.rmir:$KEY \\\n",
+    "        /servicemaker-dev/$ACOUSTIC_MODEL_NAME:$KEY \\\n",
+    "        --decoding_language_model_arpa=/servicemaker-dev/$LM_ARPA_MODEL_NAME \\\n",
+    "        --decoding_vocab=/servicemaker-dev/$DECODER_VOCAB_NAME \\\n",
+    "        --wfst_tokenizer_model=/servicemaker-dev/$WFST_TOKENIZER \\\n",
+    "        --wfst_verbalizer_model=/servicemaker-dev/$WFST_VERBALIZER \\\n",
+    "        --name=arpa_ngram_lm_pipeline \\\n",
+    "        --decoder_type=flashlight \\\n",
+    "        --chunk_size=0.16 \\\n",
+    "        --padding_size=1.92 \\\n",
+    "        --ms_per_timestep=80 \\\n",
+    "        --flashlight_decoder.asr_model_delay=-1 \\\n",
+    "        --vad.residue_blanks_at_start=-2 \\\n",
+    "        --featurizer.use_utterance_norm_params=False \\\n",
+    "        --featurizer.precalc_norm_time_steps=0 \\\n",
+    "        --featurizer.precalc_norm_params=False "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Riva-deploy\n",
+    "\n",
+    "The deployment tool takes as input one or more Riva Model Intermediate Representation (RMIR) files and a target model repository directory. It creates an ensemble configuration specifying the pipeline for the execution and finally writes all those assets to the output model repository directory.\n",
+    "\n",
+    "If you built an `.rmir` file using a `.binary`- or `.arpa`-formatted n-gram language model file, change `asr_riva_ngram_lm` in the cell below to `asr_binary_ngram_lm` or `asr_arpa_ngram_lm` as appropriate. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Syntax: riva-deploy -f dir-for-rmir/model.rmir:key output-dir-for-repository\n",
+    "! docker run --rm --gpus 0 -v $MODEL_LOC:/data $RIVA_SM_CONTAINER -- \\\n",
+    "            riva-deploy -f  /data/asr_riva_ngram_lm.rmir:$KEY /data/models/"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "## Start the Riva Server\n",
+    "After the model repository is generated, we are ready to start the Riva server. First, download the [Riva Quick Start](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/riva/resources/riva_quickstart) resource from NGC. \n",
```

Comment:
Just a clarification, this link is for data center platforms.  Embedded platforms is a different link.  Just wanted to clarify.

Reply:
Yes, that's true. I thought about maybe adding some words on Riva Embedded Quick Start, but Nikhil said not to bother.